### PR TITLE
T22070 Emit ::invalidated when a schedule entry is removed

### PR DESCRIPTION
**Pull request by pwithnall**
https://phabricator.endlessm.com/T22070

### CODE REVIEW

File: libmogwai-schedule/schedule-service.c [hunk 1]
```diff
@@ -510,6 +510,26 @@ entries_changed_cb (MwsScheduler *scheduler,
                         (GCallback) entry_notify_cb, self);
     }
 
+  /* Emit Removed signals for entries. */
+  for (gsize i = 0; removed != NULL && i < removed->len; i++)
+    {
+      MwsScheduleEntry *entry = MWS_SCHEDULE_ENTRY (removed->pdata[i]);
+      g_autoptr(GError) local_error = NULL;
+
```
Comment (joaquimrocha) [inline libmogwai-schedule/schedule-service.c]: I'd say there's no need for this newline but I don't mind if you did this on purpose.

File: libmogwai-schedule/schedule-service.c [hunk 2]
```diff
@@ -510,6 +510,26 @@ entries_changed_cb (MwsScheduler *scheduler,
...
+      g_dbus_connection_emit_signal (self->connection,
+                                     mws_schedule_entry_get_owner (entry),
+                                     entry_path,
+                                     "com.endlessm.DownloadManager1.ScheduleEntry",
+                                     "Removed",
+                                     NULL,  /* no arguments */
+                                     &local_error);
+      if (local_error != NULL)
+        g_debug ("Error emitting Removed signal for ‘%s’: %s",
+                 local_error->message, entry_path);
```
Comment (joaquimrocha) [inline libmogwai-schedule/schedule-service.c]: Shouldn't this be reversed? i.e. the `entry_path` should be the first format arg, right?

File: libmogwai-schedule/schedule-service.c (hunk 1 above)
Comment (pwithnall) [reply libmogwai-schedule/schedule-service.c]: It was on purpose.

File: libmogwai-schedule/schedule-service.c (hunk 2 above)
Comment (pwithnall) [reply libmogwai-schedule/schedule-service.c]: Ooooops, yes. Well spotted.

**Comment by pwithnall** [comment]
> Also, should there be unit tests for this?

There should, but there isn’t a unit testing harness for this code yet. That’s https://phabricator.endlessm.com/T20423, which is getting higher up my list.

File: libmogwai-schedule/schedule-service.c (hunk 2 above)
Comment (pwithnall) [reply libmogwai-schedule/schedule-service.c]: fixup pushed.

**Comment by joaquimrocha** [comment]
LGTM now. Please fixup and merge.

**Comment by pwithnall** [comment]
Rebased, squashed and merged; thanks.

**Pull request closed without merging**